src/Decompiler/Analysis/IndirectCallRewriter.cs

```diff
@@ -0,0 +1,213 @@
+    {
+        private SsaState ssa;
+        private Frame frame;
+        private Statement stm;
+        private CallInstruction call;
+
+        public SsaIdentifierTransformer(SsaState ssa)
+        {
+            this.ssa = ssa;
+            this.frame = ssa.Procedure.Frame;
+        }
+
+        public void Transform(Statement stm, CallInstruction call)
+        {
+            this.stm = stm;
+            this.call = call;
+            ssa.ReplaceDefinitions(stm, null);
+            ssa.RemoveUses(stm);
+            Transform(stm.Instruction);
+        }
+
+        public override Instruction TransformAssignment(Assignment a)
+        {
+            a.Src = a.Src.Accept(this);
+            a.Dst = FindDefinedId(call, a.Dst.Storage);
+            if (a.Dst == null)
+                return new SideEffect(a.Src);
+            DefId(a.Dst, a.Src);
+            return a;
+        }
+
+        public override Expression VisitApplication(Application appl)
+        {
+            appl.Procedure = appl.Procedure.Accept(this);
+            for (int i = 0; i < appl.Arguments.Length; ++i)
+                appl.Arguments[i] = TransformArgument(appl.Arguments[i]);
+            return appl;
+        }
+
+        private Expression TransformArgument(Expression arg)
+        {
+            var id = arg as Identifier;
+            if (id == null)
+                return arg;
+            var usedId = FindUsedId(call, id.Storage);
+            if (usedId != null)
+                UseId(usedId);
+            return usedId ?? InvalidArgument();
+        }
+
+        private Expression InvalidArgument()
+        {
+            return Constant.Word32(Constant.Invalid.ToUInt32());
+        }
+
+        public override Expression VisitIdentifier(Identifier id)
+        {
+            UseId(id);
+            return id;
+        }
+
+        private void DefId(Identifier id, Expression defExp)
+        {
+            SsaIdentifier sid;
+            if (ssa.Identifiers.TryGetValue(id, out sid))
+            {
+                sid.DefExpression = defExp;
+                sid.DefStatement = stm;
+            }
+        }
+
+        private void UseId(Identifier id)
+        {
+            SsaIdentifier sid;
+            if (ssa.Identifiers.TryGetValue(id, out sid))
+            {
+                sid.Uses.Add(stm);
+            }
+        }
+
+        private Identifier FindDefinedId(CallInstruction call, Storage storage)
+        {
+            return call.Definitions.Select(d => d.Expression).
+                OfType<Identifier>().
+                Where(usedId => usedId.Storage.Equals(storage)).
+                FirstOrDefault();
+        }
+
+        private Identifier FindUsedId(CallInstruction call, Storage storage)
+        {
+            var locStorage = storage as StackLocalStorage;
+            // $HACK add return address size to stack offset to correct
+            // parameters binding. Is it correct?
+            storage =
```

Comment:
STDAPI? What do you mean? The passing of arguments in `stdcall` the same as in `cdecl` (the only difference is clearing stack)